rest-api-spec/src/main/resources/rest-api-spec/api/text_structure.find_field_structure.json

```diff
@@ -0,0 +1,103 @@
+{
+  "text_structure.find_field_structure":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/find-field-structure.html",
+      "description":"Finds the structure of a text field in an index."
+    },
+    "stability":"stable",
+    "visibility":"public",
+    "headers":{
+      "accept": ["application/json"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/_text_structure/find_field_structure",
+          "methods":["GET"]
+        }
+      ]
+    },
+    "params":{
+      "index":{
+        "type":"string",
+        "description":"The index containing the analyzed field",
+        "required":true
+      },
+      "field":{
+        "type":"string",
+        "description":"The field that should be analyzed",
+        "required":true
+      },
+      "lines_to_sample":{
+        "type":"int",
+        "description":"How many documents should be included in the analysis",
+        "default":1000
+      },
+      "line_merge_size_limit":{
+        "type":"int",
+        "description":"Maximum number of characters permitted in a single message when lines are merged to create messages.",
+        "default":10000
+      },
+      "timeout":{
+        "type":"time",
+        "description":"Timeout after which the analysis will be aborted",
+        "default":"25s"
+      },
+      "charset":{
+        "type":"string",
+        "description":"Optional parameter to specify the character set of the file"
+      },
+      "format":{
+        "type":"enum",
+        "options":[
+          "ndjson",
+          "xml",
+          "delimited",
+          "semi_structured_text"
+        ],
+        "description":"Optional parameter to specify the high level file format"
+      },
+      "has_header_row":{
+        "type":"boolean",
+        "description":"Optional parameter to specify whether a delimited file includes the column names in its first row"
+      },
```

Comment:
This also seems inappropriate for an endpoint that's working on fields that contain one record from a delimited file. It's likely that for this endpoint `column_names` should be compulsory _if_ the data is delimited.